aviary/mission/flops_based/phases/two_dof_phase.py

```diff
@@ -0,0 +1,267 @@
+        initial_ref = user_options.get_val('initial_ref', units='ft')
+        duration_bounds = user_options.get_val('duration_bounds', units='ft')
+        duration_ref = user_options.get_val('duration_ref', units='ft')
+        ground_roll = user_options.get_val('ground_roll')
+        balance_throttle = user_options.get_val('balance_throttle')
+        rotation = user_options.get_val('rotation')
+        constraints = user_options.get_val('constraints')
+        optimize_mach = user_options.get_val('optimize_mach')
+        optimize_altitude = user_options.get_val('optimize_altitude')
+        mach_bounds = user_options.get_item('mach_bounds')
+        altitude_bounds = user_options.get_item('altitude_bounds')
+
+        if not balance_throttle:
+            phase.add_parameter(
+                Dynamic.Mission.THROTTLE,
+                opt=False,
+                units="unitless",
+                val=throttle_setting,
+                static_target=False)
+
+        if fix_initial:
+            phase.set_time_options(fix_initial=fix_initial, fix_duration=False,
+                                   units='ft', name=Dynamic.Mission.DISTANCE,
+                                   duration_bounds=duration_bounds, duration_ref=duration_ref)
+        else:
+            phase.set_time_options(fix_initial=fix_initial, fix_duration=False,
+                                   units='ft', name=Dynamic.Mission.DISTANCE,
+                                   initial_bounds=initial_bounds, initial_ref=initial_ref,
+                                   duration_bounds=duration_bounds, duration_ref=duration_ref)
+
+        phase.set_state_options("time", rate_source="dt_dr",
+                                fix_initial=fix_initial, fix_final=False, ref=100., defect_ref=100. * 1.e2, solve_segments='forward')
+
+        phase.set_state_options("mass", rate_source="dmass_dr",
+                                fix_initial=fix_initial, fix_final=False, ref=170.e3, defect_ref=170.e5,
+                                val=170.e3, units='lbm', lower=10.e3)
+
+        phase.add_polynomial_control(Dynamic.Mission.MACH,
+                                     order=control_order,
+                                     val=0.4, units=mach_bounds[1],
+                                     opt=optimize_mach, lower=mach_bounds[0][0], upper=mach_bounds[0][1],
+                                     fix_initial=fix_initial,
+                                     rate_targets=['dmach_dr'],
+                                     )
+
+        if rotation:
+            phase.add_polynomial_control("alpha",
+                                         order=control_order,
+                                         fix_initial=True,
+                                         lower=0, upper=15,
+                                         units='deg', ref=10.,
+                                         val=0.,
+                                         opt=opt)
+
+        if ground_roll:
```

Comment:
Can you have groundroll in the 2dof? I thought setting groundroll to True gave us the Groundroll phase instead.

Reply:
These names are mildly confusing; I'll clean this up in my follow-on PRs